libraries/AP_GPS/AP_GPS_MAV.cpp

```diff
@@ -42,67 +42,107 @@ bool AP_GPS_MAV::read(void)
 // corresponding gps data appropriately;
 void AP_GPS_MAV::handle_msg(const mavlink_message_t *msg)
 {
-    mavlink_gps_input_t packet;
-    mavlink_msg_gps_input_decode(msg, &packet);
-
-    bool have_alt    = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_ALT) == 0);
-    bool have_hdop   = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_HDOP) == 0);
-    bool have_vdop   = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VDOP) == 0);
-    bool have_vel_h  = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VEL_HORIZ) == 0);
-    bool have_vel_v  = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VEL_VERT) == 0);
-    bool have_sa     = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_SPEED_ACCURACY) == 0);
-    bool have_ha     = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_HORIZONTAL_ACCURACY) == 0);
-    bool have_va     = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VERTICAL_ACCURACY) == 0);
-
-    state.time_week     = packet.time_week;
-    state.time_week_ms  = packet.time_week_ms;
-    state.status = (AP_GPS::GPS_Status)packet.fix_type;
-
-    Location loc = {};
-    loc.lat = packet.lat;
-    loc.lng = packet.lon;
-    if (have_alt) {
-        loc.alt = packet.alt;
+    switch (msg->msgid) {
+
+        case MAVLINK_MSG_ID_GPS_INPUT: {
+            mavlink_gps_input_t packet;
+            mavlink_msg_gps_input_decode(msg, &packet);
+
+            bool have_alt    = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_ALT) == 0);
+            bool have_hdop   = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_HDOP) == 0);
+            bool have_vdop   = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VDOP) == 0);
+            bool have_vel_h  = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VEL_HORIZ) == 0);
+            bool have_vel_v  = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VEL_VERT) == 0);
+            bool have_sa     = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_SPEED_ACCURACY) == 0);
+            bool have_ha     = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_HORIZONTAL_ACCURACY) == 0);
+            bool have_va     = ((packet.ignore_flags & GPS_INPUT_IGNORE_FLAG_VERTICAL_ACCURACY) == 0);
+
+            state.time_week     = packet.time_week;
+            state.time_week_ms  = packet.time_week_ms;
+            state.status = (AP_GPS::GPS_Status)packet.fix_type;
+
+            Location loc = {};
+            loc.lat = packet.lat;
+            loc.lng = packet.lon;
+            if (have_alt) {
+                loc.alt = packet.alt;
+            }
+            state.location = loc;
+            state.location.options = 0;
+
+            if (have_hdop) {
+                state.hdop = packet.hdop * 100; // convert to centimeters
```

Comment:
Pedantic, but this is unit less, it's not in centimeters, not sure how pervasive the problem is, but since you had to tab the line in... :) (same with all other dop's)